Tests/SeedsTest.m

```diff
@@ -0,0 +1,221 @@
+    appKey = @"1234";
+    appHost = @"www.testurl.com";
+    deviceId = nil;
+    seedsMock = OCMClassMock([Seeds class]);
+}
+
+- (void)tearDown {
+    [super tearDown];
+}
+
+- (void) testStart {
+    [[Seeds sharedInstance] start:appKey withHost: appHost];
+    
+    XCTAssertEqual(appKey, [[Seeds sharedInstance] getAppKey]);
+    XCTAssertTrue([[Seeds sharedInstance] isStarted]);
+    XCTAssertNotEqual(deviceId, [[Seeds sharedInstance] deviceId]);
+}
+
+- (void) testStartWithDeviceId {
+    deviceId = @"Iphone 5s";
+    [[Seeds sharedInstance] start:appKey withHost:appHost andDeviceId:deviceId];
+    
+    XCTAssertEqual(appKey, [[Seeds sharedInstance] getAppKey]);
+    XCTAssertEqual(appKey, [[SeedsInterstitialAds sharedInstance] appKey]);
+    XCTAssertEqual(appKey, [[SeedsConnectionQueue sharedInstance] appKey]);
+    XCTAssertEqual(appHost, [[SeedsConnectionQueue sharedInstance] appHost]);
+    XCTAssertEqual(appHost, [[SeedsInterstitialAds sharedInstance] appHost]);
+
+    XCTAssertTrue([[Seeds sharedInstance] isStarted]);
+    XCTAssertNotNil([[Seeds sharedInstance] deviceId]);
+    XCTAssertEqual(deviceId, [[Seeds sharedInstance] deviceId]);
+}
+
+// Mock this method to test properly
+- (void) testStartWithTestMessagingUsing {
+    [[Seeds sharedInstance] startWithTestMessagingUsing:appKey withHost:appHost andOptions:nil];
+    
+    XCTAssertEqual(appKey, [[Seeds sharedInstance] getAppKey]);
+    XCTAssertTrue([[Seeds sharedInstance] isStarted]);
+    XCTAssertNotEqual(deviceId, [[Seeds sharedInstance] deviceId]);
+    XCTAssertTrue([[SeedsConnectionQueue sharedInstance] startedWithTest]);
+}
+
+- (void) testSeedsNotificationCategories {
+    NSMutableSet *categories = [[Seeds sharedInstance] seedsNotificationCategories];
+    
+    XCTAssertNotNil(categories);
+    XCTAssertEqual(3, categories.count);
+}
+
+- (void) testSeedsNotificationCategoriesWithActionTitles {
+    NSArray *actions = @[@"Open", @"Update", @"Review", @""];
+    
+    NSMutableSet *categories = [[Seeds sharedInstance] seedsNotificationCategoriesWithActionTitles:actions];
+    
+    XCTAssertNotNil(categories);
+    XCTAssertEqual(3, categories.count);
+}
+
+- (void) testRecordEvent {
+    [seedsMock recordEvent:appKey count:2];
+    OCMVerify([seedsMock recordEvent:appKey count:2]);
+    
+    [seedsMock recordEvent:appKey count:3 sum:10];
+    OCMVerify([seedsMock recordEvent:appKey count:3 sum:10]);
+    
+    [seedsMock recordEvent:appKey segmentation:nil count:1];
+    OCMVerify([seedsMock recordEvent:appKey segmentation:nil count:1]);
+    
+    [seedsMock recordEvent:appKey segmentation: nil count:4 sum:2];
+    OCMVerify([seedsMock recordEvent:appKey segmentation: nil count:4 sum:2]);
+}
+
+- (void) testRecordUserDetails {
+    [seedsMock recordUserDetails:@{}];
+    OCMVerify([seedsMock recordUserDetails:@{}]);
+}
+
+- (void) testTrackPurchase {
+    [seedsMock trackPurchase:appKey price:0.99];
+    OCMVerify([seedsMock trackPurchase:appKey price:0.99]);
```

Comment:
Are this kind of "call a function, check if the function" tests useful? Writing trivial tests is sometimes really good, but are these tests even too trivial? Or is the idea just to make it explicit when the method signature changes?